Mage.Sets/src/mage/cards/f/Fiendlash.java

```diff
@@ -0,0 +1,161 @@
+package mage.cards.f;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.DealtDamageToSourceTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.keyword.ReachAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AttachmentType;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.DamagedEvent;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetPlayerOrPlaneswalker;
+
+/**
+ *
+ * @author zeffirojoe
+ */
+public final class Fiendlash extends CardImpl {
+
+    public Fiendlash(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.ARTIFACT }, "{1}{R}");
+
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +2/+0 and has reach.
+        Ability staticAbility = new SimpleStaticAbility(new BoostEquippedEffect(2, 0));
+        staticAbility.addEffect(new GainAbilityAttachedEffect(ReachAbility.getInstance(), AttachmentType.EQUIPMENT)
+                .setText("and has reach"));
+        this.addAbility(staticAbility);
+
+        // Whenever equipped creature is dealt damage, it deals damage equal to its
+        // power to target player or planeswalker.
+        this.addAbility(new FiendlashTriggeredAbility());
+
+        // Equip {2}{R}
+        this.addAbility(new EquipAbility(Outcome.AddAbility, new ManaCostsImpl<>("{2}{R}")));
+    }
+
+    private Fiendlash(final Fiendlash card) {
+        super(card);
+    }
+
+    @Override
+    public Fiendlash copy() {
+        return new Fiendlash(this);
+    }
+}
+
+class FiendlashTriggeredAbility extends TriggeredAbilityImpl {
+
+    private boolean usedForCombatDamageStep = false;
+
+    FiendlashTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new FiendlashEffect(), false);
+        this.addTarget(new TargetPlayerOrPlaneswalker());
+    }
+
+    private FiendlashTriggeredAbility(final FiendlashTriggeredAbility ability, boolean usedForCombatDamageStep ) {
+        super(ability);
+        this.usedForCombatDamageStep = usedForCombatDamageStep;
+    }
+
+    @Override
+    public FiendlashTriggeredAbility copy() {
+        return new FiendlashTriggeredAbility(this, false);
```

Comment:
Copy must use copy constructor only. Copy constructor must have only one final param. That's rules for any xmage classes. Example:
* `FiendlashTriggeredAbility(final FiendlashTriggeredAbility ability`).

If you need params/settings in constructor then create the new one with it. Also you must copy all additional settings (e.g. `usedForCombatDamageStep`) in copy constructor code like:
```
FiendlashTriggeredAbility(final FiendlashTriggeredAbility ability`) {
    super(ability);
    this.usedForCombatDamageStep = ability.usedForCombatDamageStep;
}
```